unit/testing-utils/get_goto_model_from_c.cpp

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************\
+
+Module: Get goto model
+
+Author: Daniel Poetzl
+
+\*******************************************************************/
+
+#include "get_goto_model_from_c.h"
+
+#include <ansi-c/ansi_c_language.h>
+
+#include <cbmc/cbmc_parse_options.h>
+
+#include <goto-programs/goto_convert_functions.h>
+
+#include <langapi/language_file.h>
+#include <langapi/mode.h>
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/invariant.h>
+#include <util/message.h>
+#include <util/options.h>
+#include <util/symbol_table.h>
+
+#include <testing-utils/message.h>
+
+namespace
+{
+bool initialized = false;
+optionst options;
+
+void initialize()
+{
+  PRECONDITION(!initialized);
+
+  register_language(new_ansi_c_language);
+
+  cmdlinet cmdline;
+
+  config.main = std::string("main");
```

Comment:
Actually, I think this shouldn’t be called only once... because `config` is a global variable and could be messed with by other tests, ideally this would start with something like `config = configt{};` so we can be sure we’re starting from a “clean” config and run every time (it’s not like this is going to cost us much test performance).

Reply:
Good point. I've now gotten rid of the initialize function altogether.